packages/multi-provider/src/utils.ts

```diff
@@ -100,7 +100,9 @@ export function evmId(data: BytesLike): Address {
  * @throws if either side is not a valid 20 or 32 byte nomad identifier
  */
 export function equalIds(left: BytesLike, right: BytesLike): boolean {
-  return canonizeId(left) === canonizeId(right);
+  const l = canonizeId(left);
+  const r = canonizeId(right);
+  return l.every((v, i) => v === r[i]);
```

Comment:
what does this fix? o.O

Reply:
@prestwich `canonizeId(...)` returns an arrayish. In JS u cannot compare 2 arrays with simple equals. Here is some more https://stackoverflow.com/questions/13523611/how-to-compare-two-arrays-in-node-js

Reply:
oh I forgot it returns a uint8array ya this is fint

Reply:
So, shall we merge?